app/models/spree/volume_price.rb

```diff
@@ -14,6 +14,16 @@ class Spree::VolumePrice < ActiveRecord::Base
 
   validate :range_format
 
+  def self.for_variant(variant, user: nil)
+    role_id = user ? user.resolve_role : nil
+
+    where(
+      (arel_table[:variant_id].eq(variant.id)
+        .or(arel_table[:volume_price_model_id].in(variant.volume_price_models.ids)))
+        .and(arel_table[:role_id].eq(role_id))
+    ).order(position: :asc)
```

Comment:
Just verifying intended behaviour here is `(variant_id || price_models.include?(model_id)) && role_id`